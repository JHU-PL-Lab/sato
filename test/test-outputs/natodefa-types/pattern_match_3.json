[
  {
    "err_errors": [
      [
        "Error_match",
        {
          "err_match_aliases": [ "a" ],
          "err_match_val": "{}",
          "err_match_expected": "Function",
          "err_match_actual": "Record {}"
        }
      ],
      [
        "Error_match",
        {
          "err_match_aliases": [ "a" ],
          "err_match_val": "{}",
          "err_match_expected": "Boolean",
          "err_match_actual": "Record {}"
        }
      ],
      [
        "Error_match",
        {
          "err_match_aliases": [ "a" ],
          "err_match_val": "{}",
          "err_match_expected": "Integer",
          "err_match_actual": "Record {}"
        }
      ]
    ],
    "err_input_seq": [],
    "err_location": "match a with | int -> 0 | bool -> 1 | fun -> 2 end",
    "err_steps": 103
  }
]
