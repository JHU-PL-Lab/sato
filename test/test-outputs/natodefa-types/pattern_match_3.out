Type errors for:
- Input sequence : []
- Found at expr  : match a with | int -> 0 | bool -> 1 | fun -> 2
end
--------------------
* Value    : a = {}
* Expected : Function
* Actual   : Record {}
--------------------
* Value    : a = {}
* Expected : Boolean
* Actual   : Record {}
--------------------
* Value    : a = {}
* Expected : Integer
* Actual   : Record {}
3 errors found. No further control flows exist.
